features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/DrawableUtils.kt

```diff
@@ -72,8 +80,47 @@ internal class DrawableUtils(
         return Pair(width, height)
     }
 
-    internal companion object {
-        internal const val MAX_BITMAP_SIZE_IN_BYTES = 10240 // 10kb
+    @Suppress("ReturnCount")
+    private fun getBitmapBySize(
+        displayMetrics: DisplayMetrics,
+        width: Int,
+        height: Int,
+        config: Config
+    ): Bitmap? {
+        var bitmap = getReusableBitmapFromPool(width, height, config)
+        if (bitmap == null) {
+            bitmap = createNewBitmap(displayMetrics, width, height, config)
+        }
+        return bitmap
+    }
+
+    private fun getReusableBitmapFromPool(
```

Comment:
why this extra method ?

Reply:
I wanted to make the intent of what was happening here clearer